diff --git a/interpreters/berry/CMakeLists.txt b/interpreters/berry/CMakeLists.txt
--- a/interpreters/berry/CMakeLists.txt
+++ b/interpreters/berry/CMakeLists.txt
@@ -37,6 +37,8 @@ if(CONFIG_INTERPRETERS_BERRY)
         ${CMAKE_CURRENT_LIST_DIR}/berry/default/berry.c ${CMAKE_CURRENT_LIST_DIR}/0001-Fix-Berry-default-port-for-NuttX.patch
         COMMAND patch -l -p1 -d ${CMAKE_CURRENT_LIST_DIR}/berry -i
         ${CMAKE_CURRENT_LIST_DIR}/0001-Fix-Berry-default-port-for-NuttX.patch
+        COMMAND patch -l -p1 -d ${CMAKE_CURRENT_LIST_DIR}/berry -i
+        ${CMAKE_CURRENT_LIST_DIR}/0002-Use-NuttX-console-reader.patch
       DOWNLOAD_NO_PROGRESS true
       TIMEOUT 30)
 
@@ -56,7 +58,8 @@ if(CONFIG_INTERPRETERS_BERRY)
   set(BERRY_CONFIG ${BERRY_CONFIG_DIR}/berry_conf.h)
   set(BERRY_GENERATE ${BERRY_DIR}/generate)
   set(BERRY_GEN_STAMP ${BERRY_GENERATE}/.generated)
-  set(BERRY_FLAGS -Wno-unused-but-set-variable -Wno-shadow)
+  set(BERRY_FLAGS -DUSE_NUTTX_CONSOLE -Wno-unused-but-set-variable
+                  -Wno-shadow)
   set(BERRY_INCDIR ${BERRY_CONFIG_DIR} ${BERRY_DIR}/src ${BERRY_DIR}/default)
 
   set(BERRY_CSRCS
diff --git a/interpreters/berry/Makefile b/interpreters/berry/Makefile
--- a/interpreters/berry/Makefile
+++ b/interpreters/berry/Makefile
@@ -63,5 +63,6 @@ CFLAGS += ${INCDIR_PREFIX}$(APPDIR)$(DELIM)interpreters$(DELIM)berry$(DELIM)incl
 CFLAGS += ${INCDIR_PREFIX}$(APPDIR)$(DELIM)interpreters$(DELIM)berry$(DELIM)$(BERRY_UNPACK)$(DELIM)src
 CFLAGS += ${INCDIR_PREFIX}$(APPDIR)$(DELIM)interpreters$(DELIM)berry$(DELIM)$(BERRY_UNPACK)$(DELIM)default
+CFLAGS += -DUSE_NUTTX_CONSOLE
 CFLAGS += -Wno-unused-but-set-variable
 CFLAGS += -Wno-shadow
 
@@ -80,6 +81,7 @@ $(BERRY_UNPACK): $(BERRY_ARCHIVE)
 	$(Q) mv berry-$(BERRY_COMMIT_ID) $(BERRY_UNPACK)
 	$(Q) perl -pi -e 's/\r$$//' $(BERRY_UNPACK)$(DELIM)default$(DELIM)berry.c 0001-Fix-Berry-default-port-for-NuttX.patch
 	$(Q) patch -l -d $(BERRY_UNPACK) -p1 < 0001-Fix-Berry-default-port-for-NuttX.patch
+	$(Q) patch -l -d $(BERRY_UNPACK) -p1 < 0002-Use-NuttX-console-reader.patch
 
 $(BERRY_GEN_STAMP): $(BERRY_UNPACK) $(BERRY_CONFIG)
 	$(Q) echo "Generating Berry constant objects"
