diff --git a/arch/arm/src/rp23xx/CMakeLists.txt b/arch/arm/src/rp23xx/CMakeLists.txt
index b305881..474541f 100644
--- a/arch/arm/src/rp23xx/CMakeLists.txt
+++ b/arch/arm/src/rp23xx/CMakeLists.txt
@@ -95,6 +95,10 @@ if(CONFIG_ADC)
   list(APPEND SRCS rp23xx_adc.c)
 endif()
 
+if(CONFIG_IEEE80211_INFINEON_CYW43439)
+  list(APPEND SRCS rp23xx_cyw43439.c)
+endif()
+
 if(CONFIG_WATCHDOG)
   list(APPEND SRCS rp23xx_wdt.c)
 endif()
diff --git a/arch/arm/src/rp23xx/Make.defs b/arch/arm/src/rp23xx/Make.defs
index 3753788..aa9d774 100644
--- a/arch/arm/src/rp23xx/Make.defs
+++ b/arch/arm/src/rp23xx/Make.defs
@@ -102,6 +102,10 @@ ifeq ($(CONFIG_ADC),y)
 CHIP_CSRCS += rp23xx_adc.c
 endif
 
+ifeq ($(CONFIG_IEEE80211_INFINEON_CYW43439),y)
+CHIP_CSRCS += rp23xx_cyw43439.c
+endif
+
 ifeq ($(CONFIG_WATCHDOG),y)
 CHIP_CSRCS += rp23xx_wdt.c
 endif
diff --git a/boards/Kconfig b/boards/Kconfig
index 918eb45..d40cb29 100644
--- a/boards/Kconfig
+++ b/boards/Kconfig
@@ -2235,6 +2235,12 @@ config ARCH_BOARD_RASPBERRYPI_PICO_2
 	---help---
 		This is a port to the Raspberry Pi Pico 2 board.
 
+config ARCH_BOARD_RASPBERRYPI_PICO_2_W
+	bool "Raspberry Pi Pico 2 W board"
+	depends on ARCH_CHIP_RP23XX
+	---help---
+		This is a port to the Raspberry Pi Pico 2 W board.
+
 config ARCH_BOARD_RASPBERRYPI_PICO_2_RV
 	bool "Raspberry Pi Pico 2 board RISC-V (not Pico, not W)"
 	depends on ARCH_CHIP_RP23XX_RV
@@ -3918,6 +3924,7 @@ config ARCH_BOARD
 	default "waveshare-rp2040-lcd-1.28"    if ARCH_BOARD_WAVESHARE_RP2040_LCD_1_28
 	default "adafruit-fruit-jam-rp2350"    if ARCH_BOARD_ADAFRUIT_FRUIT_JAM_RP2350
 	default "raspberrypi-pico-2"           if ARCH_BOARD_RASPBERRYPI_PICO_2
+	default "raspberrypi-pico-2-w"         if ARCH_BOARD_RASPBERRYPI_PICO_2_W
 	default "raspberrypi-pico-2-rv"        if ARCH_BOARD_RASPBERRYPI_PICO_2_RV
 	default "xiao-rp2350"                  if ARCH_BOARD_XIAO_RP2350
 	default "pimoroni-pico-2-plus"         if ARCH_BOARD_PIMORONI_PICO_2_PLUS
@@ -4445,6 +4452,9 @@ endif
 if ARCH_BOARD_RASPBERRYPI_PICO_2
 source "boards/arm/rp23xx/raspberrypi-pico-2/Kconfig"
 endif
+if ARCH_BOARD_RASPBERRYPI_PICO_2_W
+source "boards/arm/rp23xx/raspberrypi-pico-2-w/Kconfig"
+endif
 if ARCH_BOARD_ADAFRUIT_FRUIT_JAM_RP2350
 source "boards/arm/rp23xx/adafruit-fruit-jam-rp2350/Kconfig"
 endif
